onedrive_service/src/onedrive_service/authentication/one_drive_client.py

```diff
@@ -0,0 +1,43 @@
+import onedrivesdk
+from onedrivesdk.helpers import GetAuthCodeServer
+
+from .session import Session
+from .base_client import BaseOneDriveClient
+
+
+API_BASE_URL = "https://api.onedrive.com/v1.0/"
+
+
+class OneDriveClient(BaseOneDriveClient):
+    """ API client for OneDrive for personal use.
+    """
+
+    def __init__(self, config: dict) -> None:
+        """ Basic initialization.
+        """
+
+        self._config = config
+        client_id = config["auth"]["client_id"]
+        scopes = config["auth"]["scopes"]
+
+        http_provider = onedrivesdk.HttpProvider()
+        auth_provider = onedrivesdk.AuthProvider(http_provider, client_id, scopes, session_type=Session)
```

Comment:
Can we sublcass `Session` and make it just save it's token?

Reply:
We don't need to care about token. Official OneDrive SDK incapculate all functionality for updating token (`AuthProvider.authenticate_request` method checks is token expired and updates it if neccesary).

Reply:
How frequently it happens?

Reply:
Token expires in 3600 seconds (1 hour). `AuthProvider` checks whether token expired before each request, and refresh this token if necessary.